en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/message-transformation/defining-interceptors-in-an-open-api-definition.md

```diff
@@ -0,0 +1,245 @@
+# Defining Interceptors in an OpenAPI Definition
+
+You can define interceptors on an API level (per API) and on a resource level (per resource).
+If you define a request/response interceptor on an API level **and** a resource level, the API level interceptor will be
+overridden by the resource level interceptor for that resource.
+
+For an API, you can define
+
+- one of request or response interceptor
+- or both of request and response interceptor
+
+base on your requirement for the mediation.
+
+
+## Request flow Interceptor
+
+You can define the request flow interceptor as follows in the Open API Definition with the extension `x-wso2-request-interceptor`.
+[Description of Keys of Interceptor Open API Extension](#description-of-keys-of-interceptor-open-api-extension) explains
+detailed description of the keys.
```

Comment:
For a detailed description of the keys, see [Description of Keys of Interceptor Open API Extension](#description-of-keys-of-interceptor-open-api-extension).